app/controllers/users_controller.rb

```diff
@@ -1,9 +1,22 @@
 class UsersController < ApplicationController
 
-    def new
+  def new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      redirect_to root_path, notice: 'Account created successfully'
+    else
+      flash[:error] = 'An error occurred!'
+      render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.